RackApp/app.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+require 'erb'
+require 'rack'
+require 'active_support/time'
+
+class MyController
+  attr_reader :person, :time
+
+  def initialize(user, time)
+    @person = user
+    @time = time
+  end
+
+  def file(format)
+    File.read(File.join(__dir__, "template.#{format}.erb"))
+  end
+
+  def template(format)
+    ERB.new(file(format))
+  end
+
+  def render(format)
+    template(format).result(binding)
+  end
+end
+
+class User
+  attr_reader :nickname
+
+  def initialize(nickname)
+    @nickname = nickname
+  end
+end
+
+class CountryTime
+  attr_reader :zone, :timezone
+
+  def initialize(zone)
+    @zone = zone
+  end
+
+  def timezone
+    Time.zone = @zone
+    @time = Time.zone.now
+  end
+end
+
+class Display
+  def initialize(app = nil)
+    @app = app
+  end
+
+  def call(env)
+    @app&.call(env)
+    req = Rack::Request.new(env)
+    format = request_format(env)
+    puts env.inspect
+    nickname = req.params['nickname']
+    tz = req.params['tz']
+    f = req.params['f']
```

Comment:
`f` is not the ideal name for a variable.. the code should be readable as a book, so the variable names should make some sense.. `timezone` instead of `tz`, `format` instead of  `f`